storage/innobase/lock/lock0wait.cc

```diff
@@ -362,6 +364,18 @@ void lock_wait_suspend_thread(
 
   if (lock_wait_timeout < 100000000 && wait_time > (double)lock_wait_timeout &&
       !trx_is_high_priority(trx)) {
+    if (lock_type == LOCK_REC) {
+      trx_set_detailed_error(
+          trx, timeout_message("record in index", wait_lock->index->table_name,
```

Comment:
This could be a bug in the original implementation, but is `wait_lock` safe to access outside of `lock_sys->mutex`?

Reply:
This is wrong, but even mutex is not enough as `wait_lock` may already be freed.